### ISSUE TITLE
Re-create VB WinForms Application property page (4)

### ISSUE THREAD

**Issue by tmeschter** [dotnet/project-system#7236]
We need to recreate the VB Application property page for WinForms projects in the new project properties system.

For reference, this is the Application property page as it currently appears for VB WinForms apps targeting .NET 5:

![image](https://user-images.githubusercontent.com/10506730/118733539-5110f800-b7f1-11eb-8b85-cd02796b6ea0.png)

Note this differs from the C# Application property page in some key ways:

- Different handling of resources
- The "Assembly Information..." button that opens a separate dialog for changing assembly information. It is not clear if this applies to .NET Core projects or only makes sense for .NET Framework projects.
- The "View Windows Settings" button to open the app.config file.
- "Enable application framework" and all the related settings. These are interesting as they may change settings in code or the app.config file rather than in the .vbproj file.

There are a couple of options for implementing this. C# and VB could share a "base" Application page with all the common settings, and then each have their own "extending" page that adds the language-specific items. Or, they could each have their own totally separate "Application" page, with a few properties duplicated. I think it really only matters if we expect there to be additional common settings that we wish to add in the future.

_This work was identified as part of #6935._

**Comment by drewnoakes** [comment]
Several of these values are stored in a `.myapp` file.

**Comment by RussKie** [comment]
The values are stored in myapp file, and AFAIK, changes at this dialog invoke an "old-style" source generator, which emits necessary code into Application.Desiger.vb.

@KlausLoeffelmann @KathleenDollard we could consider moving to the Roslyn source generators (similar to https://github.com/dotnet/winforms/issues/4909), as well as moving away from myapp file in favour of vbproj MSBuild properties. 
Then maintaining the VS propoerty page experience wouldn't be as daunting experience as it is today.
Thoughts?

**Comment by paul1956** [comment]
As long as we can do a 1 time upgrade it is a fantastic idea, if we can't do it automatically some instructions on what a developer needs to do is an OK option. Right now in Core there is come confusion on how some of the properties should even be set like StartupObject, setting it wrong loses the StartUp screen for example and default Icon does nothing on this screen. What does it even mean in Core to turn off Enable XP visual styles. All the old properties that were in "Assembly Information..." are used in SplashScreen form and About Form are still hard to discover, I believe some are on package screen and stored in vbproj file but they don't populate "My" and some don't appear anywhere, for example.
```
<AssemblyVersion>5.0.3.4</AssemblyVersion>
<FileVersion>5.0.3.4</FileVersion>
<Version>5.0.3.4</Version>
```
I believe the first 2 can be set under "package version", the third don't be set anywhere but I don't know how any of them get to application.

Also VB/VS used to do an auto version increase by setting last 2 numbers to * that appears to no longer work.

**Bottom line there needs to be 1 source of truth and right now there are multiple and adding another "Assembly Information..." screen is not going to help with that.**

**Comment by Nukepayload2** [comment]
> The "View Windows Settings" button to open the app.config file.

The "View Windows Settings" button is used to open or create the "My Project\app.manifest" file.

**Comment by paul1956** [comment]
The "View Windows Settings" creates a file that refers to .Net Framwork 4.6 and Windows XP and no useful information about what to change. There is also a list of OS versions that application is tested on starting with Windows Vista (Does core work on Vista). There is a TargetFramework in project file and how does that map to this file?

**Comment by RussKie** [comment]
@paul1956 manifests are a Windows thing (e.g. used for dpi, and by common controls). Have a look at https://docs.microsoft.com/windows/win32/sbscs/application-manifests.

**Comment by paul1956** [comment]
@RussKie yes but .Net Core only runs on a subset of Windows Version why is the file being created not simplified to only support the things .Net Core Supports. Right now almost _everything_ is commented out. Is there anything a user can actually change? The DPI section seems like the only possibility and if it can be changed it is controlled by some other setting.

**Comment by RussKie** [comment]
AFAIK there is no reason to create a manifest file, unless you needs an option that is not included/enabled in the [default manifest](https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/src/Resources/System/Windows/Forms/XPThemes.manifest) Windows Forms SDK ships. And if you're creating one - then we trust you know what you're doing.

**Comment by melytc** [comment]
Note: this should be enabled for .NET 6 or higher projects.
